### ISSUE TITLE
Build error(s) for gcc 6.1.1

### ISSUE THREAD

**Issue by SleuthKid** [01org/hyperscan#33]
We encountered some build errors with the current stable version of hyperscan 4.2.0 after Debian switched to gcc 6.1.1 lately:

```
[ 17%] Building CXX object CMakeFiles/hs_shared.dir/src/hs.cpp.o
/usr/bin/c++   -Dhs_shared_EXPORTS -I/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu -I/<<PKGBUILDDIR>> -I/<<PKGBUILDDIR>>/src -isystem /<<PKGBUILDDIR>>/include  -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fPIE -fstack-protector-strong -Wformat -Werror=format-security -mssse3 -march=core2 -Wdate-time -D_FORTIFY_SOURCE=2  -O2 -std=c++11 -Wall -Wextra -Wno-shadow -Wswitch -Wreturn-type -Wcast-qual -Wno-deprecated -Wnon-virtual-dtor -Werror -fabi-version=0 -Wno-unused-local-typedefs -Wno-maybe-uninitialized -fno-omit-frame-pointer -Wvla -Wpointer-arith -Wno-unused-const-variable -Wmissing-declarations -fPIC   -o CMakeFiles/hs_shared.dir/src/hs.cpp.o -c /<<PKGBUILDDIR>>/src/hs.cpp
In file included from /<<PKGBUILDDIR>>/src/util/charreach.h:40:0,
                 from /<<PKGBUILDDIR>>/src/nfagraph/ng_graph.h:41,
                 from /<<PKGBUILDDIR>>/src/nfagraph/ng_holder.h:32,
                 from /<<PKGBUILDDIR>>/src/nfagraph/ng.h:36,
                 from /<<PKGBUILDDIR>>/src/hs.cpp:40:
/<<PKGBUILDDIR>>/src/util/bitfield.h:420:38: error: ignoring attributes on template argument 'ue2::bitfield<requested_size>::block_type {aka long long unsigned int}' [-Werror=ignored-attributes]
     std::array<block_type, num_blocks> bits;
                                      ^
In file included from /<<PKGBUILDDIR>>/src/nfagraph/ng.h:45:0,
                 from /<<PKGBUILDDIR>>/src/hs.cpp:40:
/<<PKGBUILDDIR>>/src/util/report_manager.h:146:23: error: ignoring attributes on template argument 's64a {aka long long int}' [-Werror=ignored-attributes]
     std::map<s64a, u32> toExhaustibleKeyMap;
                       ^
cc1plus: all warnings being treated as errors
CMakeFiles/hs_shared.dir/build.make:94: recipe for target 'CMakeFiles/hs_shared.dir/src/hs.cpp.o' failed
```

Building was done with the latest gcc 6.1.1 in Debian unstable:

```
gcc (Debian 6.1.1-11) 6.1.1 20160802
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

I will test the current development branch to see if the problem will be relevant for 4.3.0

**Comment by SleuthKid** [comment]
The problem seems to be sbuild related. sbuild sets -DCMAKE_BUILD_TYPE=None which seems to enable -Werror. A workaround is to explicitly set -DCMAKE_BUILD_TYPE=RelWithDebInfo

As gcc 6.1.1 throws a lot of new warning for the master and development branch this led to the compile errors.

**Comment by mdb256** [comment]
Thanks for the report. I've already got an upcoming commit to push that disables specific warnings on GCC 6, but we haven't accounted for using a build type of "None" turning `-Werror` on.

"None" appears to be mentioned in older CMake documentation, not so much post-v3.0, but this should be a simple change.

**Comment by mdb256** [comment]
We have disabled the warning in e3c9bc7 and fixed the other compile flag issues in 85e2ba7 - both are part of v4.3.0
